src/python/pants/jvm/dependency_inference/artifact_mapper.py

```diff
@@ -243,10 +243,39 @@ class AllJvmTypeProvidingTargets(Targets):
     pass
 
 
+class AllJvmArtifactPackagesTargets(Targets):
+    pass
+
+
+class JvmArtifactPackagesMapping(FrozenDict[UnversionedCoordinate, FrozenOrderedSet[str]]):
+    @classmethod
+    def from_dict(
+        cls, data: Mapping[UnversionedCoordinate, OrderedSet[str]]
+    ) -> JvmArtifactPackagesMapping:
+        return JvmArtifactPackagesMapping(
+            {key: FrozenOrderedSet(value) for key, value in data.items()}
+        )
+
+
 @rule(desc="Find all jvm_artifact targets in project", level=LogLevel.DEBUG)
 def find_all_jvm_artifact_targets(targets: AllTargets) -> AllJvmArtifactTargets:
     return AllJvmArtifactTargets(
-        tgt for tgt in targets if tgt.has_fields((JvmArtifactGroupField, JvmArtifactArtifactField))
+        tgt
+        for tgt in targets
+        if tgt.has_fields((JvmArtifactGroupField, JvmArtifactArtifactField, JvmResolveField))
```

Comment:
Added the `JvmResolveField` field to filter out `jvm_artifact_packages` and it is actually being used and assumed to exist on the target.